### DISCUSSION TITLE
Uso del separatore in HA

### DISCUSSION THREAD

**Discussion by ganzerlimino** [indomus/forum#503] [HUB: Home Assistant]
Ciao a tutti, dopo una settimana alzo bandiera bianca e chiedo aiuto...
Ho usato come spunto il tutorial "DIY: Monitor Your Car Battery: Code & Setup" che ho trovato qui: 

**LINK RIMOSSO**

Della macchina poco me ne frega, lo volevo usare per monitorare la batteria dell'unico UPS senza scheda di rete che ho.
Ho fatto tutta la parte sotto Tasmota e tutta la parte di Home Assistant, ovvero aggiungere in sensor.yaml il seguente :

```
  - platform: mqtt
    unique_id: wemos_battery_001
    state_topic: "tasmota/tele/wemos_battery_001/SENSOR"
    name: UPS Soffitta
    unit_of_measurement: "v"
    value_template: "{{ (value_json.ANALOG.Range | float / 100 | round(2) )}}"
    qos: 2
```

Il valore che mi viene mostrato, però, è con il punto delle migliaia, non ha la virgola per i due decimali.
Al posto di mostrare 13,94v viene visualizzato 1.394
Se qualcuno mi potesse mostrare dove diavolo sto sbagliando gliene sarei molto grato. 

Se poi avete un link che spieghi decentemente l'uso del "value_template" sarebbe il massimo :-)

**Comment by snakuzzo** [comment]
Non sono ammessi link ad altri siti

**Comment by snakuzzo** [comment]
Infatti è giusto così. Non si usa la virgola come separatore, ma il punto

**Reply by ganzerlimino** [reply]
La cosa mi può stare anche bene, ma se il valore "grezzo" è 1394, non ho modo di avere un separatore a due decimali oppure ho solo il punto delle migliaia? Nel progetto che avevo erroneamente linkato, il valore viene presentato con la virgola e due decimali. Mi chiedevo appunto come si ottenesse tale risultato. Ma se è giusto così, me ne sto.

**Reply by snakuzzo** [reply]
Pensa solo al punto come separatore decimale. Il separatore delle migliaia...dimenticalo, non ti serve

**Reply by snakuzzo** [reply]
considera che il valore restituito da un template è sempre una stringa...e ti basta dividere per 100

![image](https://user-images.githubusercontent.com/7205371/125169796-19e51480-e1ac-11eb-89d7-85104611a2fa.png)

**Reply by ganzerlimino** [reply]
Grazie mille, mi hai puntato nella direzione che avevo solo sfiorato! Non velo l'ora di tornare a casa e dedicarmici! Mille grazie!

**Reply by ganzerlimino** [reply]
Appare evidente che c'è ancora qualcosa che mi sfugge...
Ho fatto la modifica, ho riavviato tutto, ma il valore resta sempre 1.395 (o similari).

Mi sono andato a prendere il messaggio MQTT ed è il seguente :

`21:05:50.699 MQT: tasmota/tele/wemos_battery_001/SENSOR = {"Time":"2021-07-10T21:05:50","ANALOG":{"Range":1395}}
`

Sia che io metta /100 oppure /10 oppure /1000 , ottengo sempre 1.395
A questo punto mi sono andato a vedere la sintassi, me la sono replicata nei template e ci ho un pò giocato.
Tutto funziona regolare come Diego mi ha mostrato.

A questo punto, dal sensor.yaml ho ridotto tutto ai minimi termini, mettendo ciò:

 `   value_template: "{{ value_json.ANALOG.Range }}"
`

Ho anche levato la definizione della tipologia di unità di misura, ovvero

`unit_of_measurement: "v"`

Il valore restituito da Lovelace resta sempre e comunque 1.395

**E' lampante che mi sto perdendo qualcosa.**
Il brutto è che non capisco cosa.
Ma nel frattempo mi sto documentanto su Jinja2, mi si è aperto un mondo, mille grazie!
Mino

**Reply by snakuzzo** [reply]
`value_template: "{{ value_json.ANALOG.Range | float / 100 }}"`

...e lascia l'unità di misura che non c'entra niente e ti servirà per i grafici

**Reply by ganzerlimino** [reply]
Anche così, purtroppo, il valore resta 1.395
Mannaggia :-(

**Reply by ganzerlimino** [reply]
Sono anche andato ad iscrivermi al topic MQTT andando in Impostazioni-Integrazioni-Mosquitto broker-Configura-Ascoltare un argomento
Come argomento ho messo 

`tasmota/tele/wemos_battery_001/SENSOR`

Il risultato è stato :

```
{
    "Time": "2021-07-11T02:30:50",
    "ANALOG": {
        "Range": 1388
    }
}
```

Esattamente quello che mi aspettavo, ma era giusto per essere doppiamente sicuro che quello che viene inviato dal dispositivo è quello che entra in HA, senza alcuna manipolazione nel mezzo...
Purtroppo per me, sono comunque punto a capo con l'obiettivo che avevo.

Giusto per sgombrare il campo da eventuali "non detti", quando vado a modificare il sensor.yaml provvedo ovviamente a riavviare il tutto facendo un bel Impostazioni-Controlli del server-Gestione del server-Riavviare

**Reply by ganzerlimino** [reply]
Giusto per completezza, non ne ho cavato un ragno dal buco, amen, mi metto l'animo in pace, studio e vedo dove arrivo.

**Reply by snakuzzo** [reply]
Tanto per...Hai verificato le opzioni di localizzazione che hai su HA ?

**Reply by ganzerlimino** [reply]
Ciao, se ti riferisci a Impostazioni-> Generale-> Unità di misura è impostato su "Metrico Celsius, chilogrammi".

Sulle prime pensavo di aver mal impostato qualcosa io sotto questo punto di vista e me lo ero andato a cercare, purtroppo senza successo... Non sono esplicitate in configuration.yaml , ma la wiki dice che non è necessario, quindi credo vada bene. Ho provato tempo fa ad esplicitarle comunque, ma senza effetto tangibile...

**Reply by snakuzzo** [reply]
Non so che dirti...
Come indicato [QUI](https://github.com/indomus/forum/discussions/503#discussioncomment-988656) ... funziona tutto

**Reply by ganzerlimino** [reply]
Posso solo dirti grazie del tempo che mi hai dedicato! Ora mi sono incaponito, metto su un altra istanza di HA come macchina virtuale e provo a tirare dentro solo quel sensore, voglio vedere dove risiede il problema :-)
Grazie ancora!

**Reply by snakuzzo** [reply]
Figurati...ma tirare su una nuova istanza non serve a nulla secondo me.
